### ISSUE TITLE
Pygame built on android give errors while opening files

### ISSUE THREAD

**Issue by T-Dynamos** [pygame/pygame#3179]
This is my code https://raw.githubusercontent.com/T-Dynamos/Python-Games/master/Connected/main.py

Here's repo 
https://github.com/T-Dynamos/Python-Games/tree/master/Connected

This is logcat error:
```
2022-05-15 22:25:30.762 7724 7748 com.pyguru.connected I python : pygame.error: Unable to open file '/data/data/com.pyguru.connected/files/app/Sounds/flip.mp3'
```
Here's the fix i tried but not worked
```
def fileR(filename):


	return os.path.join(os.path.dirname(os.path.abspath(__file__)),filename)
```
After this function
```
flip_fx = pygame.mixer.Sound(fileR('Sounds/flip.mp3'))
```
After adding
```
import os

try:
	print(os.listdir("/data/data/com.pyguru.connected/files/app/Sounds/"))
	print(os.path.exists("/data/data/com.pyguru.connected/files/app/Sounds/flip.mp3"))
except Exception as e:
	print(str(e))
``` 
To code i got this
```
2022-05-15 23:22:31.687 29920 29942 com.pyguru.connected I python : ['bgm.mp3', 'dead.mp3', 'flip.mp3', 'point.mp3', 'score_page.mp3']
2022-05-15 23:22:31.688 29920 29942 com.pyguru.connected I python : True
```
Logcat
![img](https://i.stack.imgur.com/bSYcF.jpg)

**Comment by T-Dynamos** [comment]
https://github.com/pyGuru123/Python-Games/pull/4#issuecomment-1126923621
Heres a test build

**Comment by CAPTAIN1947** [comment]
You have to use absolute paths in android and not the way you are doing it... get the absolute path using os.path.abspath(".") and add it to the relative path of the file you want to open... also be sure to edit your buildozer.spec to include .mp3 files. We had discussed a lot about this in the past in discord. Be sure to check the android-dev channel there if you use discord.

**Comment by T-Dynamos** [comment]
Now rebuilding with
```

def fileR(filename):


	return os.path.abspath(".")+filename
```

**Comment by T-Dynamos** [comment]
No success still same and please give discord server link

**Comment by Starbuck5** [comment]
```py
return os.path.abspath(".")+filename
```
This wouldn't even work on desktop.

Gives you something like `C:\\Users\\charl\\DesktopSounds/flip.mp3`

By "add it" CAPTAIN probably meant using os.path.join.

Discord: https://discord.gg/kD2Qq9tbKm

**Comment by Pixelsuft** [comment]
```py
return os.path.join(os.getcwd(), filename)
```
worked for me

**Comment by Starbuck5** [comment]
On discord we talked about it, op hasn't responded in a few days here or three so I'm going to assume the problem is solved and close the issue.

**Comment by T-Dynamos** [comment]
Sorry , Check latest build 
https://github.com/T-Dynamos/Python-Games/actions/runs/2357668983
With
```
print(os.path.exists(fileR('Sounds/flip.mp3')))
```

**Comment by Starbuck5** [comment]
I don't know how to handle this.

**Comment by T-Dynamos** [comment]
Please reopen

**Comment by Starbuck5** [comment]
Why does your code look like this: https://github.com/T-Dynamos/Python-Games/blob/master/Connected/main.py?

I'm convinced this an issue on your end, because CAPTAIN was able to do it. It's just failing on mixer because that happens before fonts or images.

**Comment by Starbuck5** [comment]
Actually, maybe whatever pygame is getting pulled into this build process doesn't have access to mp3 functionality.

I just saw that os.path.exists was returning True for you on discord.

Try using ogg or wav instead

**Comment by CAPTAIN1947** [comment]
I manually checked and so yes the .mp3 files are in the correct place in the installed system after installing the apk. And I also checked his implementation which looks correct to me.  So, the problem is with format(.mp3) ig. Do as @Starbuck5 asked i.e try with .wav instead.

**Comment by CAPTAIN1947** [comment]
And about the "include_pattern" thing I never use them as by default buildozer should include all the files and as it seems it does so. I never used it but as the build was failing I just asked @T-Dynamos to try that. Buildozer had a recent release and I thought maybe it changed then.

**Comment by CAPTAIN1947** [comment]
Also, p4a has a lot of problems... you shouldn't hope that everything will work out of the box... like pygame is built without freetype and portmidi. Also, the recipe includes a number of patches iirc. And the main thing, when I last checked the SDL version in p4a was 2.0.9 or something(sadly its not trivial to update the SDL version)... so all these things combined  may create problems. 
Maybe, @Starbuck5 can confirm if the dated SDL version is the issue...

**Comment by T-Dynamos** [comment]
https://github.com/T-Dynamos/Python-Games/actions/runs/2363060035
So now i am trying to load images first to confirm images work 
One more thing in Pydroid3 game is working with mp3 as it also uses p4a

**Comment by T-Dynamos** [comment]
> Why does your code look like this: https://github.com/T-Dynamos/Python-Games/blob/master/Connected/main.py?
> 
> I'm convinced this an issue on your end, because CAPTAIN was able to do it. It's just failing on mixer because that happens before fonts or images.
> 
> This error means FileNotFoundError, see #2792
> 
> I see you also have a bulldozer.spec in Snake, does that one pull in the splash images it wants successfully?
> 
> Interestingly, the "include_patterns" option either doesn't exist or isn't documented in the bulldozer docs. At least it's not here: https://buildozer.readthedocs.io/en/latest/specifications.html

Now check i fixed the raw file and removed fileR()

**Comment by CAPTAIN1947** [comment]
> Now check i fixed the raw file and removed fileR()
So its working now? @T-Dynamos

**Comment by T-Dynamos** [comment]
No , wait let me see logs

**Comment by T-Dynamos** [comment]
![Uploading Screenshot_2022-05-21-18-25-47-231_com.conena.logcat.reader.jpg…]()
As i said unable to open any `files` now it does not even load files

**Comment by T-Dynamos** [comment]
> I manually checked and so yes the .mp3 files are in the correct place in the installed system after installing the apk. And I also checked his implementation which looks correct to me. So, the problem is with format(.mp3) ig. Do as @Starbuck5 asked i.e try with .wav instead.

Not about music files , it's unable to open any file

**Comment by Starbuck5** [comment]
@T-Dynamos Do you ever try to include your image files?

**Comment by T-Dynamos** [comment]
No i tried building pygame first time previously i have built kivy and tkinter apps

**Comment by Starbuck5** [comment]
@T-Dynamos

You moved the images to load first, and they don't load, so now you say it can't load any file?
I don't think that's accurate. You never try to include the pngs or assets/* files in bulldozer.spec

**Comment by T-Dynamos** [comment]
Buildozer includes all the files in cwd 
```
# (list) Source files to include (let empty to include all the files)
#source.include_exts = py,png,jpg,kv,atlas

```

**Comment by T-Dynamos** [comment]
![IMG_20220522_164548](https://user-images.githubusercontent.com/68729523/169692526-5a205412-1ed1-450f-ae4c-822d17214892.jpg)
Logs in pydroid3 says that it uses Python 3.9.7 ,pygame =2.0.1,sdl = 2.0.12
But when i tried to specify sdl = 2.0.12 it says `Could not find any version from none`

**Comment by T-Dynamos** [comment]
Now i am trying building with @CAPTAIN1947's technique to load file 
https://github.com/T-Dynamos/Python-Games/commit/e16eacdc15391fce706f567624dd9116d679048d

**Comment by Starbuck5** [comment]
> Buildozer includes all the files in cwd

Yes, leaving it empty specifies everything. But does it do everything recursively? You want things in folders that are in cwd, not things in cwd itself.

**Comment by T-Dynamos** [comment]
Folders are also included in cwd , i have made some applications in kivy for that i never specificed the Assets or Sounds folder it just includes itself
